client/tests/core/test_pattern.py

```diff
@@ -49,7 +48,7 @@ def test_program():
         assert job.in_terminal_state()
         assert job.status() == "DONE"
 
-        recovered_job = serverless.get_job_by_id(job.job_id)
+        recovered_job = serverless.get_job(job.job_id)
```

Comment:
I would agree to use the new method in the test but I would like to maintain backward compatibility for the old method too, @korgan00 . As mentioned above.

Reply:
@korgan00 correct me if I'm wrong but this problem is not fixed, isn't it? I couldn't find where we are supporting the old reference to `get_job_by_id `

Reply:
I sent a couple of messages showing the place where the deprecated functions are.
Additionally, the `test/basic/05_retrieving_past_resuilts.py` uses `get_job_by_id`, that is interesting because it is tested.